frontend/src/Menu/menu.js

```diff
@@ -0,0 +1,14 @@
+import { CHANGE_DIAPLAYED_COMPONENT } from './actionCreator';
+
+const initialState = {
+  currentlyDisplayComponent: 'Buildings',
+};
+
+export default function menu(state = initialState, action) {
+  switch (action.type) {
+    case CHANGE_DIAPLAYED_COMPONENT:
+      return { ...state, currentlyDisplayComponent: action.payload };
+    default:
+      return { ...state, currentlyDisplayComponent: 'Buildings' };
```

Comment:
```suggestion
      return state;
```
every reducer will run after if any action is dispatched, this is why we create switch-cases, so we can handle the proper action. This is why in the default in every reducer, we always return the state, and nothing else. Don't modify the state in the default.

also please rename the file to menuReducer.js or just reducer.js (because other reducers will be in different folders)